cpp/src/arrow/util/bitmap_ops.cc

```diff
@@ -311,39 +345,49 @@ Result<std::shared_ptr<Buffer>> BitmapAnd(MemoryPool* pool, const uint8_t* left,
                                           int64_t left_offset, const uint8_t* right,
                                           int64_t right_offset, int64_t length,
                                           int64_t out_offset) {
+  // TODO(pradeep) Figure out the return type for this version of BitmapOps
+  int64_t out_pop_count = 0;
   return BitmapOp<std::bit_and>(pool, left, left_offset, right, right_offset, length,
-                                out_offset);
+                                out_offset, &out_pop_count);
```

Comment:
I don't understand why this `out_pop_count`. It looks useless.

Reply:
Wanted to ask the arrow devs if the functions returning class objects also need to be modified to return pop count. It is an incomplete change ( I though I left this PR in draft mode, hm). Please share your thoughts - if you don't want to change these versions of bitmap API calls or otherwise. Hence left the TODO line above this function call https://github.com/apache/arrow/pull/11674/files#diff-1162624f731f4758d9b900e493a4551a266ea06d081f7ade43a21e333a63d8c0R348

Reply:
I think it's better to return popcount for both versions.

Reply:
In the function that is returning output value, should I make it pair of int64_t and Array ? what is usually done in arrow ?

Reply:
I think we can make the api same as the void func. Adding an output int64 argument with default value nullptr is okay.

Reply:
okay, it looks odd though :). So about `nullptr`, the linter is throwing up about it, May I use `NULL` ?

Reply:
`NULLPTR` is okay.